onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -198,23 +202,16 @@ def _preprocess(self, in_mp):
 
         # returns True if no more ready nodes
         def _insert_ready_nodes():
-            ready_nodes = [pn for pn in pending_nodes if all([i in defined for i in pn.input if i])]
+            ready_nodes = [pn for pn in pending_nodes if len(pn.input) == 0 or all([i in defined for i in pn.input if i])]
```

Comment:
>len(pn.input) == 0  [](start = 57, length = 19)

nit:

if not pn.input

Reply:
Thanks, rewritten this part

---
In reply to: [487200955](https://github.com/microsoft/onnxruntime/pull/5107#discussion_r487200955) [](ancestors = 487200955)